storage/memory/log_storage.go

```diff
@@ -342,6 +342,22 @@ func (t *logTreeTX) GetActiveLogIDs(ctx context.Context) ([]int64, error) {
 	return t.getActiveLogIDs(ctx)
 }
 
+func (t *readOnlyLogTX) GetUnsequencedCounts(ctx context.Context) (map[int64]int64, error) {
+	t.ms.mu.RLock()
+	defer t.ms.mu.RUnlock()
+
+	ret := make(map[int64]int64)
+	for id, tree := range t.ms.trees {
+		tree.mu.RLock()
```

Comment:
nit: Use `tree.RLock()` (or remove it altogether?). Ditto for the other `tree.*Lock()` methods.

(pre-existing) nit: I would expect the global `t.ms.mu` lock to be enough for an in-memory impl. If it's not, a comment explaining why would be helpful.